pages/instructorScanPaper/instructorScanPaper.js

```diff
@@ -0,0 +1,80 @@
+const express = require('express');
+const router = express.Router();
+const path = require('path');
+const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
+
+const { processScrapPaperPdf } = require('../../lib/scrapPaperReader');
+const serverJobs = require('../../lib/server-jobs');
+
+const error = require('../../prairielib/lib/error');
+const ERR = require('async-stacktrace');
+
+router.get('/', (req, res) => {
+  res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+});
+
+router.post('/', function (req, res, next) {
+  if (!res.locals.authz_data.has_course_instance_permission_edit) return next();
+  if (req.body.__action === 'scan_scrap_paper') {
+    if (!req.file) {
+      return next(error.make(400, 'Missing barcoded pdf collection file data'));
+    }
+    if (!res.locals || !res.locals.authn_user || !res.locals.authn_user.user_id) {
+      return next(error.make(400, 'Authn_user required on file-store API'));
```

Comment:
This error message is pretty opaque. Can you improve it?

Reply:
Done.